### ISSUE TITLE
List of predefined caveats

### ISSUE THREAD

**Issue by bwegh** [paulmillar/macaroon-minting#2]
it would be great to have a list of predefined caveats,
although I do not know if that is out of scope of this document

**Comment by paulmillar** [comment]
I was wondering about that, too.

My take is that a "caveat discover protocol" would be a separate protocol, possibly using the same endpoint, but with a different Content-Type (e.g., application/json+macaroon-caveat-discovery).

I need to update the macaroon-minting protocol to allow other requests on the same endpoint.

**Comment by bwegh** [comment]
yes the discover protocol is not a bad idea, maybe the discover protocol endpoint should be the one specified in the minting protocol and the minting endpoint can be anywhere specified in the json document returned at the discover endpoint (like OpenID Connect).

**Comment by bwegh** [comment]
still we need to define a list of predefined caveats and how to verify them, as the discovery won't be able to describe what a caveat means.
I assume the discovery returns something like "supported_caveats: exp, uid, url, x509"
what do those caveats mean, how can they be used? etc ... there should be a first list of caveats including how to verify them, so macaroon libraries can implement them.

**Comment by paulmillar** [comment]
Exactly.  This "can of worms" is why I didn't want to specify anything about caveat semantics in the minting protocol.  "Somehow" the client knows what caveats to write there.

**Comment by bwegh** [comment]
okay, I see your point.
Still I think this "can of worms" is worth a document on its own.

**Comment by paulmillar** [comment]
Agreed!  ... but lets try to get this document out first :-)
